### ISSUE TITLE
Establish and enforce minimum supported Rust version policy

### ISSUE THREAD

**Issue by danielocfb** [libbpf/libbpf-rs#262]
We should try and determine the minimum supported Rust version and enforce it by using it in CI as well. If it's sufficiently low, we may consider bumping it so that we can $$ in on new features and [bump](https://github.com/libbpf/libbpf-rs/pull/261) to using Edition 2021.

**Comment by danielocfb** [comment]
Test runs on a bunch of Rust versions:

https://github.com/danielocfb/libbpf-rs/actions/runs/2974040910

We seem to already have an example that is using Edition 2021. Open question whether that should carry over to the entire library, though.

**Comment by chenhengqi** [comment]
The README already states that:

> WARNING: The API is not stable and is subject to breakage.

I guess it's OK to apply to the entire library.

**Comment by danielocfb** [comment]
> The README already states that:
> 
> > WARNING: The API is not stable and is subject to breakage.

Agreed, but I don't think that's to be read as a blanket statement for all sorts of breakages whenever we feel like it. Buying into the `crates.io` eco-system we almost certainly should follow semantic versioning (or [Cargo's version](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio) of it). Depending on what we decide, that may or may not [include](https://doc.rust-lang.org/cargo/reference/semver.html#env-new-rust) the minimum supported Rust version (MSRV). In any event, long term we almost certainly want to err on the side of stability and for that having enforcement of MSRV and intentional bumps is certainly a plus, if not a necessity.
